src/test/bridges/example/Example.t.sol

```diff
@@ -14,60 +14,118 @@ contract ExampleTest is BridgeTestBase {
+        bridge = new ExampleBridgeContract(address(ROLLUP_PROCESSOR));
 
         // use the label cheatcode to mark the address with "Example Bridge" in the traces
-        vm.label(address(exampleBridge), "Example Bridge");
+        vm.label(address(bridge), "Example Bridge");
 
         // Impersonate the multi-sig to add a new bridge
         vm.prank(MULTI_SIG);
 
         // List the example-bridge with a gasLimit of 100K
-        ROLLUP_PROCESSOR.setSupportedBridge(address(exampleBridge), 100000);
+        ROLLUP_PROCESSOR.setSupportedBridge(address(bridge), 100000);
 
         // Fetch the id of the example bridge
         id = ROLLUP_PROCESSOR.getSupportedBridgesLength();
     }
 
-    function testErrorCodes(address _callerAddress) public {
+    function testInvalidCaller(address _callerAddress) public {
         vm.assume(_callerAddress != address(ROLLUP_PROCESSOR));
         vm.prank(_callerAddress);
         vm.expectRevert(ErrorLib.InvalidCaller.selector);
-        exampleBridge.convert(emptyAsset, emptyAsset, emptyAsset, emptyAsset, 0, 0, 0, address(0));
+        bridge.convert(emptyAsset, emptyAsset, emptyAsset, emptyAsset, 0, 0, 0, address(0));
     }
 
-    function testExampleBridge(uint256 _depositAmount) public {
-        uint256 depositAmount = bound(_depositAmount, 2, type(uint96).max);
+    function testInvalidInputAssetType() public {
+        vm.prank(address(ROLLUP_PROCESSOR));
+        vm.expectRevert(ErrorLib.InvalidInputA.selector);
+        bridge.convert(emptyAsset, emptyAsset, emptyAsset, emptyAsset, 0, 0, 0, address(0));
+    }
+
+    // @notice The purpose of this test is to directly test convert functionality of the bridge.
+    // @dev In order to avoid overflows we set _depositAmount to be uint96 instead of uint256.
+    function testExampleBridgeUnitTest(uint96 _depositAmount) public {
+        // Define input and output assets
+        AztecTypes.AztecAsset memory inputAssetA = AztecTypes.AztecAsset({
+            id: 1,
+            erc20Address: address(DAI),
+            assetType: AztecTypes.AztecAssetType.ERC20
+        });
+
+        // Rollup processor transfers ERC20 tokens to the bridge before calling convert. Since we are calling
+        // bridge.convert(...) function directly we have to transfer the funds in the test on our own. In this case
+        // we'll solve it by directly minting the _depositAmount of Dai to the bridge.
+        deal(address(DAI), address(bridge), _depositAmount);
+
+        // Store dai balance before interaction to be able to verify the balance after interaction is correct
+        uint256 daiBalanceBefore = DAI.balanceOf(address(ROLLUP_PROCESSOR));
+
+        // Impersonate rollup processor
+        vm.startPrank(address(ROLLUP_PROCESSOR));
+
+        // Disabling linting errors here to show return variables
+        // solhint-disable-next-line
+        (uint256 outputValueA, uint256 outputValueB, bool isAsync) = bridge.convert(
```

Comment:
Would make an assertion for each of the output values, just to make it clear that we expect that